src/base/iana/rcode.rs

```diff
@@ -854,3 +854,103 @@ impl From<OptRcode> for TsigRcode {
 }
 
 int_enum_str_with_decimal!(TsigRcode, u16, "unknown TSIG error");
+
+//============ Tests =========================================================
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn optrcode_parts() {
+        // RFC 1035 defines a 4-bit RCODE in the main DNS header RCODE field.
```

Comment:
Perhaps this text should be in the doc string of OptRcode?

Reply:
Is this better?

Reply:
Perfect!